client/src/components/molecules/DonationItemForm.tsx

```diff
@@ -0,0 +1,307 @@
+            }
+            errorString = errorString.concat(" " + error);
+        });
+        errorString = errorString.concat(".");
+        setFormError(errorString);
+
+        return errorString;
+    };
+
+    const onConditionChange = (newCondition: ItemCondition) => {
+        setCondition(newCondition);
+        setIsConditionDropdownOpen(false);
+        updateFormError(newCondition, quantity);
+    };
+
+    const onAgeChange = (newAge: number) => {
+        setAge(newAge);
+        setIsAgeDropdownOpen(false);
+    };
+
+    const onQuantityChange = (newQuantity: number) => {
+        setQuantity(newQuantity);
+        updateFormError(condition, newQuantity);
+    };
+
+    const onDescriptionChange = (newDescription: string) => {
+        setDescription(newDescription);
+    };
+
+    const onSave = () => {
+        const nameError = updateNameError(name);
+        const formError = updateFormError(condition, quantity);
+
+        if (nameError.length === 0 && formError.length === 0) {
+            const itemRequestGroup = props.requestGroups.find((requestGroup) => requestGroup.name === name);
+
+            props.onSave({
+                age,
+                condition: condition ?? undefined,
+                description,
+                name,
+                quantity,
+                requestGroup: itemRequestGroup ?? null
+            });
+        }
+    };
+
+    return (
+        <div className="donation-item-form">
+            <FormItem
+                className="item-name-field"
+                formItemName="What item do you wish to donate?"
+                errorString={nameError}
+                isDisabled={false}
+                showErrorIcon={false}
+                inputComponent={
+                    <SearchableDropdown
+                        selectedItem={name}
+                        searchString={nameInput}
+                        dropdownItems={props.requestGroups.reduce(
+                            (acc, requestGroup) => (requestGroup.name ? acc.concat(requestGroup.name) : acc),
+                            [] as Array<string>
+                        )}
+                        initialText={nameInput}
+                        isDisabled={false}
+                        isErroneous={nameError.length !== 0}
+                        actionOption={
+                            nameInput.length === 0 ? null : (
+                                <div onClick={() => onNameChange(nameInput)}>
+                                    <span>{nameInput}</span>
+                                    <span className="custom-item-prompt">- enter your own item</span>
+                                </div>
+                            )
+                        }
+                        onChange={setNameInput}
+                        onSelect={onNameChange}
+                        placeholderText="Enter item name"
+                        searchPlaceholderText=""
```

Comment:
just some thoughts: if the user types something and clicks the "Enter" key the item name won't be updated, and if they click out of the dropdown then what they typed is lost

the designs don't specify the behaviour but is this something we need to address?

Reply:
> if they click out of the dropdown then what they typed is lost

Yeah this is some behaviour that the designers wanted from searchable dropdowns for request groups.
See [this design spec](https://www.notion.so/uwblueprintexecs/Create-Request-Modals-e7f6a33170d243638d77727ff93a3b10#ed042466093c465797801b17fe2be2b3) for request forms

> the designs don't specify the behaviour for hitting "Enter" but is this something we need to account for?

oh yes that's a good idea! i'll trying implementing this. fingers crossed it isn't had